src/wallet.rs

```diff
@@ -0,0 +1,146 @@
+use crate::consensus::SaitoMessage;
+use crate::crypto::{SaitoPublicKey, SaitoPrivateKey, Keypair};
+use ::std::{sync::Arc, thread::sleep, time::Duration};
+use tokio::sync::{broadcast, mpsc, RwLock};
+
+#[derive(Clone, Debug)]
+pub enum WalletMessage {
+    TestMessage,
+    TestMessage2,
+}
+
+/// The `Wallet` manages the public and private keypair of the node and holds the
+/// slips that are used to form transactions on the network.
+pub struct Wallet {
+
+    broadcast_channel_sender: Option<broadcast::Sender<SaitoMessage>>,
+    wallet_channel_sender: Option<mpsc::Sender<WalletMessage>>,
+
+    keypair: Keypair,
+
+}
+
+impl Wallet {
+
+    pub fn new() -> Self {
+        Wallet {
+            broadcast_channel_sender: None,
+            wallet_channel_sender: None,
+	    keypair: Keypair::new(),
+        }
+    }
+
+    pub fn get_publickey(&self) -> SaitoPublicKey {
+        self.keypair.get_publickey()
+    }
+
+    pub fn get_privatekey(&self) -> SaitoPrivateKey {
+        self.keypair.get_privatekey()
+    }
+
+    pub fn set_broadcast_channel_sender(&mut self, bcs : broadcast::Sender<SaitoMessage>) {
+      self.broadcast_channel_sender = Some(bcs);
+    }
+    pub fn set_wallet_channel_sender(&mut self, wcs : mpsc::Sender<WalletMessage>) {
+      self.wallet_channel_sender = Some(wcs);
+    }
+
+}
+
+
+
+
+
+
+
+
+
+//
+// This function is called on initialization to setup the sending
+// and receiving channels for asynchronous loops or message checks
+//
+pub async fn run(
```

Comment:
I'm confused why this function is needed or why wallet needs a loop or a recv().